### PR TITLE
Add TPM measurement support

### PR DESCRIPTION
**Pull request by insomniacslk**
Replaces https://github.com/systemboot/systemboot/pull/70 , where I fucked up @zaolin 's commits :(

### CODE REVIEW

File: pkg/crypto/measure.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	// BootConfig type in PCR 8
+	BootConfig uint32 = 8
+	// ConfigData type in PCR 8
+	ConfigData uint32 = 8
+	// NvramVars type in PCR 9
+	NvramVars uint32 = 9
+)
+
+// TryMeasureBootConfig measures bootconfig contents
+func TryMeasureBootConfig(name string, kernel string, initramfs string, kernelArgs string, deviceTree string) {
```
Comment (Natolumin) [inline pkg/crypto/measure.go]: nit: Wouldn't it be easier if this function was in bootconfig instead compared to having to unpack the struct to use it ? everything it uses is exported

**Review by Natolumin — DISMISSED** [review]
Code looks reasonable, I'm not sure about the PCR numbers; I'd like @zaolin to have a look in any case

**Comment by insomniacslk** [comment]
> Code looks reasonable, I'm not sure about the PCR numbers; I'd like @zaolin to have a look in any case

This is @zaolin 's original PR (#70), I just recreated it after I messed up a merge conflict

**Comment by insomniacslk** [comment]
in other words - not my code. I will review it anyway once @zaolin has confirmed that this matches what he originally meant

**Comment by zaolin** [comment]
I can confirm

File: pkg/crypto/measure.go (hunk 1 above)
Comment (insomniacslk) [reply pkg/crypto/measure.go]: I agree with you, this would probably better live in the bootconfig package

**Comment by insomniacslk** [comment]
So there are three things:
1. @Natolumin 's advice on moving TryMeasureBootconfig to the `bootconfig` package
2. I'm wondering whether the measurement functions would be better suited to be part of `tpmtool`
3. could we reuse an open TPM rather than opening and closing it every time? This means making the measurements function as methods of a struct

CC @zaolin for further advice

**Comment by insomniacslk** [comment]
@Natolumin , I think we can progress and merge this, and discuss on a separate issue. Could you please re-approve?

**Review by Natolumin — APPROVED** [review]

**Pull request merged**